glossary/Currying.md

```diff
@@ -0,0 +1,4 @@
+### Currying
+
+Currying is a way of constructing functions that allows partial application of a function's arguments.
```

Comment:
Depending o definition this could be a contentious point, but it's close enough to the real world... I wonder which way we should go? Actual currying(mathematical and CS: series of functions that accept a single argument that delays evaluation) or practical currying(what we do now)

Reply:
It's a decently universal definition that works for most contexts. I think it won't be an issue, plus we can leave this open for discussion as we usually do when the community starts picking up the pace and suggesting edits.